gap/WreathProductElements.gi

```diff
@@ -31,8 +31,47 @@ function(g, G)
     return ForAll([1 .. Length(l) - 1], i -> l[i] in K) and l[Length(l)] in H;
 end);
 
-# InstallMethod( \in, "perm, and perm wreath product", true,[IsObject, IsPermGroup and HasWreathProductInfo], OVERRIDENICE + 42, WPE_IN);
-# InstallMethod( \in, "matrix, and matrix wreath product", true, [IsMatrix, IsMatrixGroup and HasWreathProductInfo], OVERRIDENICE + 42, WPE_IN);
+InstallMethod( \in, "perm, and perm wreath product", true,[IsPerm, IsPermGroup and HasWreathProductInfo], OVERRIDENICE + 42,
+function(g, G)
+    local info, comps, n, i;
+    info := WreathProductInfo(G);
+    comps := ComponentsOfWreathProduct(G);
+    n := NrMovedPoints(comps[2]);
+    if not IsBound(info.projection) or Length(info.embeddings) < n + 1 then
```

Comment:
For future generations reading and debugging this code I would strongly recommend to add some comments explaining what is going on here

Reply:
yes, you are right. I will do it

Reply:
@fingolfin, I hope the comment explains the bug and fix well enough. I am thankful for any suggestions on how to improve the comment